Fix powerpc and powerpc64, where clang defines __POWERPC__ but rejects
Darwin's "r27" syntax with "error: invalid operand for instruction".

https://github.com/boostorg/fiber/commit/4d776f7e5ddf7821f8e3541a6ab7bc36ed00932b

Index: boost/fiber/detail/cpu_relax.hpp
--- boost/fiber/detail/cpu_relax.hpp.orig
+++ boost/fiber/detail/cpu_relax.hpp
@@ -59,7 +59,7 @@ namespace detail {
 //               processors
 // extended mnemonics (available with POWER7)
 // yield   ==   or 27, 27, 27
-# if defined(__POWERPC__) // Darwin PPC
+# if defined(__APPLE__) // Darwin PPC
 # define cpu_relax() asm volatile ("or r27,r27,r27" ::: "memory");
 # else
 # define cpu_relax() asm volatile ("or 27,27,27" ::: "memory");
